internal/namespace-utilities/ns-pruner.go

```diff
@@ -0,0 +1,115 @@
+package namespace_utilities
```

Comment:
Can we call this package `utilities` instead of `namespace-|_utilities`? And then rename it `namespace_pruner.go`.

If we ever need to do more "utilities" we can just extend this package then.